include/tvm/auto_scheduler/cost_model.h

```diff
@@ -70,6 +70,11 @@ class CostModelNode : public Object {
     LOG(FATAL) << "Not implemented";
   }
 
+  /*!
+   * \brief Default virtual destructor
+   */
+  virtual ~CostModelNode() {}
```

Comment:
The default v-destructor is not necessary for the object, because they are created by make_object, and the destructor is stored there. 

The `Wdelete-non-virtual-dtor` already covers that case and will create a warning if an object is created using other methods (e.g. new) but deleted via virtual destructor. Right now the code base is free of that warning.